File: solutions/prolog/darts/1/darts.pl
score(X, Y, 0):-
    radius(X,Y,Radius),
    Radius > 10.

score(X, Y, 1):-
    radius(X,Y,Radius),
    Radius > 5,
    Radius =< 10.

score(X, Y, 5):-
    radius(X,Y,Radius),
    Radius > 1,
    Radius =< 5.

score(X, Y, 10):-
    radius(X,Y,Radius),
    Radius =< 1.

radius(X, Y, Radius):-
    Radius is sqrt(X**2 + Y**2).

File: solutions/ocaml/darts/1/darts.ml
let score (x: float) (y: float): int =
  if sqrt (x*.x +. y*.y) > 10. then
    0
  else if sqrt(x*.x +. y*.y) > 5. then
    1
  else if sqrt(x*.x +. y*.y) > 1. then
    5
  else
    10
